### PR TITLE
DS-424 remove button deprecations

### PR DESCRIPTION
**Pull request by remydenton**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-424

## Summary

Removes deprecated props from button component

## Details

Removes the following props from button: `itemAlignment`, `rounded`, `tag`, `style`, `onClick`, and `onClickTarget`.

Also removes support for the values `left` and `right` in the `align` prop.

## How to test

Confirm that there are no more references to the removed props in the Bolt codebase.

## Release notes

The following props have been removed from the `button` component:
- `itemAlignment`: renamed to `align`
- `rounded`: renamed to `border_radius`
- `tag`: renamed to `type`
- `style`: renamed to `hierarchy`
- `onClick`: renamed to `on_click`
- `onClickTarget`: renamed to `on_click_target`

Additionally, the button `align` prop no longer supports `left` and `right`.  Use `start` and `end` instead.

### CODE REVIEW

File: docs-site/src/components/hamburger-button/hamburger-button.twig [hunk 1]
```diff
@@ -5,7 +5,7 @@
         text: 'Open Menu',
         iconOnly: true,
         url: "#nav",
-        style: "primary",
+        hierarchy: "primary",
         size: "medium",
         rounded: "rounded",
```
Comment (colbytcook) [inline docs-site/src/components/hamburger-button/hamburger-button.twig:10]: Should we rename "rounded" to "border_radius"?

File: docs-site/src/components/hamburger-button/hamburger-button.twig [hunk 2]
```diff
@@ -21,7 +21,7 @@
         text: 'Close Menu',
         iconOnly: true,
         url: "#",
-        style: "primary",
+        hierarchy: "primary",
         size: "medium",
         rounded: "rounded",
```
Comment (colbytcook) [inline docs-site/src/components/hamburger-button/hamburger-button.twig:26]: Should we rename "rounded" to "border_radius"

**Comment by colbytcook** [comment]
I searched through the file system and found two more Bolt Button "Tag"

- docs-site/src/pages/pattern-lab/_patterns/40-components/button/999-button-with-web-component.twig
- docs-site/src/pages/pattern-lab/_patterns/50-pages/99999-bolt-dev-sandbox/_wip-global-search.twig

These might not be real examples but i figured I should link them here anyways,

**Review by danielamorse — APPROVED** [review]
Approved, but let's wait until tomorrow morning to merge so we avoid potential conflicts with Marcin's work.

**Review by cjwhitedev — APPROVED** [review]

**Pull request merged**